tests/test_cli.py

```diff
@@ -0,0 +1,167 @@
+from pathlib import Path
+
+import click.testing
+import pytest
+
+from csv2docx import cli
+
+
+default_outpath = Path.cwd() / "output"
+
+
+@pytest.fixture
+def runner():
+    return click.testing.CliRunner()
+
+
+# not using a tmp_path to test default output folder
+def test_basic_succeeds(runner):
+    result = runner.invoke(
+        cli.main,
+        [
+            "--template",
+            "tests/data/example.docx",
+            "--data",
+            "tests/data/example.csv",
+            "--name",
+            "NAME",
+        ],
+    )
+    assert result.exit_code == 0
+
+    # remove standard output files
+    for f in default_outpath.glob("*.docx"):
+        try:
+            f.unlink()
+        except OSError as e:
+            print("Error: %s : %s" % (f, e.strerror))
+
```

Comment:
Couple of points:

- Use an f-string instead of string formatting (cleaner);
- Add `default_outpath.rmdir()` after the for loop to delete the `output` dir
- Remove the print statement and raise an assert instead since running pytests won't output the print statement, but an assert will.

This also raises an interesting question worth discussing around writing to/from files in tests, although arguments against this tend to be for projects that have many tests and therefore writing to disk slows down running tests [see here](https://endler.dev/2018/go-io-testing/). This is unlikely to impact us 👍 

The alternative is creating temporary files (as you've done below) or using _mocking_ with [unittest.mock](https://docs.python.org/3/library/unittest.mock.html). I think our current approach (reading/writing files) is fine (more functional testing than unit though?) and no need to use mocks as it'll introduce complexity that's not worth the effort imho.

Reply:
Got it, I've rewritten it as a f-string. It was a shameless act of copy-paste from an online solution.
I originally tested with `default_outpath.rmdir()` but it would error due to a non-empty folder (it contained a `.DS_Store`), but I've adjusted the code to completely empty the file.

Regarding the discussion point, the cleanup is only needed for the standard output, as the `tmp_path` and `tmpdir` are automatically cleaned  - they are kept for a t most 3 tests invocations. see [here](https://docs.pytest.org/en/latest/tmpdir.html).   So I do think we might not need the fixture. 

Regarding the unittests vs functional tests, yes, I would like to test whether the output is correct (e.g. the nameing of the .docx) but didn't got to it, so that would be more of a unittest don't you think?

Reply:
I was thinking that when we test the library by invoking `convert` directly that it would also create the `/output/<files>` and need the same cleanup. We can sort that out once we write tests for the library, but imho in a different issue since you have set the infrastructure in place here so we can progress type annotations etc alongside testing.

Yes, you're right about it being a unit test and great point about checking the output names of files. We should unit test `create_unique_name` specifically to verify naming, but of course we might want to test the default arguments/logic of the convert method also

Reply:
Yes, could point about that. For now, I've set up the cleanup as fixture to also get rid of a very pressing `flake8-aaa` error ;-) 

I'll migrate these suggestions to another issue to expand on our tests.